### PR TITLE
docs: fix spelling errors

### PR DESCRIPTION
**Pull request by notmandatory**
### Description

I did a global spell check and found and fixed a few spelling errors. 

### Notes to the reviewers

This is low priority but want to make sure it's done before we do a beta release.

### Changelog notice

None.

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

**Review by danielabrozzoni — COMMENTED** [review]
Looks good to me, can you rebase to pick up the CI fixes?

**Comment by nondiremanuel** [comment]
I put it in alpha.3 milestone to indicate that ideally it needs to be fixed within beta version

**Comment by evanlinjin** [comment]
Time I install a spellchecker :stuck_out_tongue_closed_eyes:

File: CHANGELOG.md [hunk 1]
```diff
@@ -449,7 +449,7 @@ final transaction is created by calling `finish` on the builder.
 #### Changed
 - Simplify the architecture of blockchain traits
 - Improve sync
-- Remove unused varaint HeaderParseFail
+- Remove unused variant HeaderParseFail
```
Comment (realeinherjar) [inline CHANGELOG.md]:
```suggestion
- Remove unused variant `HeaderParseFail`
```
I think that this should be in `code`.

Comment (notmandatory) [reply CHANGELOG.md]: Fixed and I added the single quotes, but in general for such an old version I wouldn't bother.

**Review by realeinherjar — APPROVED** [review]
ACK 94f8fa5

**Review by evanlinjin — APPROVED** [review]
ACK 94f8fa530b34ae1c099db3df070000121e8283cf

**Pull request merged**